azure-go-sdk-conceptual/azure-sdk-go-authorization.md

```diff
@@ -21,15 +21,18 @@ The Azure SDK for Go offers several different types of authentication, using dif
 | Authentication type | Recommended when... |
 |---------------------|---------------------|
 | Certificate-based authentication | You have an X509 certificate that was configured for an Azure Active Directory (AAD) user or service principal. To learn more, see [Get started with certificate-based authentication in Azure Active Directory]. |
-| Service principal credentials | You have a configured service principal that is set up for this application or a class of applications it belongs to. To learn more, see [Create a service principal with Azure CLI 2.0]. |
+| Client credentials | You have a configured service principal that is set up for this application or a class of applications it belongs to. To learn more, see [Create a service principal with Azure CLI 2.0]. |
 | Managed Service Identity (MSI) | Your application is running on an Azure resource that has been configured with Managed Service Identity (MSI). To learn more, see [Managed Service Identity (MSI) for Azure resources]. |
 | Device token | Your application is meant to be used interactively __only__ and will have a variety of users, potentially from multiple AAD tenants. Users have access to a web browser to log in. For more information, see [Use device token authentication](#use-device-token-authentication).|
 | Username/password | You have an interactive application that cannot use any other authentication method. Your users do not have multi-factor authentication enabled for their AAD login. |
 
 > [!IMPORTANT]
-> If you use an authentication type other than service principal credentials, your application must be registered in Azure Active Directory. To learn how,
+> If you use an authentication type other than client credentials, your application must be registered in Azure Active Directory. To learn how,

```

Comment:
Not sure what this means. you still need to have the service principal attached to an AAD application right ? Maybe you are reffering to username password ?

Reply:
As far as I can tell, a service principal acts as its own full credential set for an application which does not need to be registered in AAD. I'm not familiar with all of the details, but a service principal seems to represent a fully standalone single-user application. Every other auth method _does_ require AAD registration for the application - that's what the `ClientID` value in the constructor seems to refer to.

Reply:
Now I noticed in the doc for it, it says service principal id, but actually client credentials also refers to the App client id that the service principal is associated with

Reply:
AAD app registration is not necessary for client credentials, though. See https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth#ClientCredentialsConfig

Reply:
I've used service principals without having to (manually) register an application in AAD. It seems like service principals auto-register the application when created and I would prefer to avoid directing users to walk through the entire "application AAD registration" process unless absolutely necessary. This is why the document calls out the manual AAD registration steps only for non-client authentication.